lambda_src/drivers/process_https.py

```diff
@@ -136,12 +137,12 @@ def process_row(
     req_results_path: str = results_path
     req_cursor: str = cursor
     req_method: str = method.upper()
-
     if json:
         req_data: Optional[bytes] = (
-            json if json.startswith('{') else dumps(parse_header_dict(json))
+            json
+            if (isinstance(json, str) or json.startswith('{'))
+            else dumps(parse_header_dict(json))
         ).encode()
-        req_headers['Content-Type'] = 'application/json'
```

Comment:
why remove this?

Reply:
We are passing content-type in the external function.

Reply:
we could, but this behavior was intentional -- it mimics the way Python's requests lib works, and it might be something our EF's currently rely on, which could break them. it's also not related to this bug fix, so should be in a separate review if you want to migrate away from it.